File: api/models/pais.ts
import { DataTypes, Sequelize } from "sequelize";
import db from '../database';


const Pais = db.define(
    "CatalogoPais",
    {
        PaisId: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true,
        },
        Nombre: {
            type: DataTypes.STRING,
            allowNull: false,
            unique: true,      
        },
        FechaEdicion: {
            type: DataTypes.DATE,
            allowNull: false,
            defaultValue: Sequelize.fn('GETUTCDATE')
        },
        Habilitado: {
            type: DataTypes.TINYINT,
            allowNull: false,
            defaultValue: 1,
        },
    },
    { timestamps: false }
);

export default Pais;

File: lib/models/pais.dart
class Pais {
  Pais({
    required this.paisId,
    required this.nombre,
    required this.fechaEdicion,
    required this.habilitado,
  });

  int paisId;
  String nombre;
  DateTime fechaEdicion;
  bool habilitado;
}

// class Pais {
//   final int paisId;
//   final String nombre;
//   final DateTime fechaEdicion;
//   final bool habilitado;

//   Pais({
//     required this.paisId,
//     required this.nombre,
//     required this.fechaEdicion,
//     required this.habilitado,
//   });

//   factory Pais.fromJson(Map<String, dynamic> json) => Pais(
//         paisId: json["PaisId"],
//         nombre: json["Nombre"],
//         fechaEdicion: DateTime.parse(json["FechaEdicion"]),
//         habilitado: json["Habilitado"],
//       );
// }
